dandischema/tests/test_metadata.py

```diff
@@ -0,0 +1,238 @@
+from hashlib import md5, sha256
+import json
+from pathlib import Path
+
+from pydantic import ValidationError
+import pytest
+
+from ..consts import DANDI_SCHEMA_VERSION
+from ..metadata import (
+    migrate_001,
+    publish_model_schemata,
+    validate,
+    validate_asset_json,
+    validate_dandiset_json,
+)
+
+METADATA_DIR = Path(__file__).with_name("data") / "metadata"
+
+
+@pytest.fixture(scope="module")
+def schema_dir(tmp_path_factory):
+    return publish_model_schemata(tmp_path_factory.mktemp("schema_dir"))
+
+
+def test_asset(schema_dir):
+    with (METADATA_DIR / "asset_001.json").open() as fp:
+        data_as_dict = json.load(fp)
+    data_as_dict["schemaVersion"] = DANDI_SCHEMA_VERSION
```

Comment:
with migrations added, I think such overwrites of schemaVersion should not be needed and not desired.  But I guess for now it is ok, but let's make that explicit

```suggestion
    # overload (here and below) schemaVersion until we support automagic schema migrations
    # under assumption that the 0.3.2 schema would be forward compatible
    data_as_dict["schemaVersion"] = DANDI_SCHEMA_VERSION
```